Assets/MixedRealityToolkit-SDK/Features/SpatialAwareness/System/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -0,0 +1,572 @@
+
+        /// <inheritdoc />
+        public int MeshTrianglesPerCubicMeter
+        {
+            get
+            {
+                return meshTrianglesPerCubicMeter;
+            }
+
+            set
+            {
+                if (IsObserverRunning)
+                {
+                    throw new System.InvalidOperationException("MeshTrianglesPerCubicMeter cannot be modified while the observer is running.");
+                }
+
+                meshTrianglesPerCubicMeter = value;
+            }
+        }
+
+        /// <inheritdoc />
+        public bool MeshRecalculateNormals { get; set; } = true;
+
+        /// <inheritdoc />
+        public SpatialMeshDisplayOptions MeshDisplayOption { get; set; } = SpatialMeshDisplayOptions.None;
+
+        /// <inheritdoc />
+        public Material MeshVisibleMaterial { get; set; } = null;
+
+        /// <inheritdoc />
+        public Material MeshOcclusionMaterial { get; set; } = null;
+
+        /// <inheritdoc />
+        public Dictionary<uint, IMixedRealitySpatialAwarenessMeshDescription> GetMeshes()
+        {
+            // todo
+            return new Dictionary<uint, IMixedRealitySpatialAwarenessMeshDescription>(0);
+        }
+
+        /// <inheritdoc />
+        public Dictionary<uint, GameObject> GetMeshObjects()
+        {
+            // todo
+            return new Dictionary<uint, GameObject>(0);
+        }
+
+#endregion Mesh Handling implementation
+
+#region Surface Finding Handling implementation
+
+        /// <inheritdoc />
+        public bool UseSurfaceFindingSystem { get; set; } = false;
+
+        /// <inheritdoc />
+        public int SurfacePhysicsLayer { get; set; } = 31;
+
+        /// <inheritdoc />
+        public int SurfacePhysicsLayerMask => 1 << SurfacePhysicsLayer;
+
+        /// <inheritdoc />
+        public float SurfaceFindingMinimumArea { get; set; } = 0.025f;
+
+        /// <inheritdoc />
+        public bool RenderFloorSurfaces { get; set; } = false;
+
+        /// <inheritdoc />
+        public Material FloorSurfaceMaterial { get; set; } = null;
+
+        /// <inheritdoc />
+        public bool RenderCeilingSurfaces { get; set; } = false;
+
+        /// <inheritdoc />
+        public Material CeilingSurfaceMaterial { get; set; } = null;
+
+        /// <inheritdoc />
+        public bool RenderWallSurfaces { get; set; } = false;
+
+        /// <inheritdoc />
+        public Material WallSurfaceMaterial { get; set; } = null;
+
+        /// <inheritdoc />
+        public bool RenderPlatformSurfaces { get; set; } = false;
+
+        /// <inheritdoc />
+        public Material PlatformSurfaceMaterial { get; set; } = null;
+
+        /// <inheritdoc />
+        public Dictionary<uint, IMixedRealitySpatialAwarenessPlanarSurfaceDescription> GetSurfaces()
+        {
+            // todo
+            return new Dictionary<uint, IMixedRealitySpatialAwarenessPlanarSurfaceDescription>(0);
```

Comment:
Do we wanna make a new dictionary every time, or modify an existing dictionary (as not all the surfaces may have changed)